src/many-types/src/lib.rs

```diff
@@ -46,6 +46,9 @@ impl<'b, C> Decode<'b, C> for Percent {
     }
 }
 
+/// Equivalent to enum VecOrSingle<T> { Single(T), Vec(Vec<T>) } in
```

Comment:
Note: I wouldn't mind an enum. There never was a goal to have idempotence between encoding and decoding.

Reply:
@hansl I think the internal representation is fine, I just added that comment for clarity on what the abstraction actually means.